### PR TITLE
fix crash during bogus reconnect on login

### PR DESCRIPTION
**Pull request by AndrewMeadows**
We've got a bug that is causing an immediate reconnect to the domain on login.  That bug is tickling another more insidious bug that causes a crash: improper teardown of physics engine.  This PR fixes the crash bug and cleans up a few other things.

### CODE REVIEW

File: libraries/physics/src/ObjectMotionState.h [hunk 1]
```diff
@@ -39,11 +39,12 @@ enum MotionStateType {
...
 const uint32_t EASY_DIRTY_PHYSICS_FLAGS = (uint32_t)(EntityItem::DIRTY_TRANSFORM | EntityItem::DIRTY_VELOCITIES |
-                                                     EntityItem::DIRTY_MASS | EntityItem::DIRTY_COLLISION_GROUP);
+                                                     EntityItem::DIRTY_MASS | EntityItem::DIRTY_COLLISION_GROUP |
+                                                     EntityItem::DIRTY_MATERIAL);
 
 // These are the set of incoming flags that the PhysicsEngine needs to hear about:
-const uint32_t DIRTY_PHYSICS_FLAGS = HARD_DIRTY_PHYSICS_FLAGS | EASY_DIRTY_PHYSICS_FLAGS |
-                                     EntityItem::DIRTY_MATERIAL | (uint32_t)EntityItem::DIRTY_PHYSICS_ACTIVATION;
+const uint32_t DIRTY_PHYSICS_FLAGS = (uint32_t)(HARD_DIRTY_PHYSICS_FLAGS | EASY_DIRTY_PHYSICS_FLAGS |
+                                                EntityItem::DIRTY_PHYSICS_ACTIVATION);
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectMotionState.h]: These changes fix a bug that Howard noticed: changes to entity restitution were not getting updated in the PhysicsEngine in a timely manner.

File: libraries/physics/src/PhysicsEngine.cpp [hunk 2]
```diff
@@ -154,11 +153,11 @@ void PhysicsEngine::deleteObjects(VectorOfMotionStates& objects) {
     }
 }
 
-// Same as above, but takes a Set instead of a Vector and ommits some cleanup operations.  Only called during teardown.
+// Same as above, but takes a Set instead of a Vector.  Should only be called during teardown.
 void PhysicsEngine::deleteObjects(SetOfMotionStates& objects) {
     for (auto object : objects) {
         btRigidBody* body = object->getRigidBody();
-        _dynamicsWorld->removeRigidBody(body);
+        removeObject(object);
```
Comment (AndrewMeadows) [inline libraries/physics/src/PhysicsEngine.cpp]: This fixes the crash on login.  I was taking a shortcut during cleanup on shutdown, but it turns out something will (incorrectly) call this at login, so we have to remove the object **properly**.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 911baee8b84b7fc95ebaf9a99a0cb0c8e72a458e available at: https://gist.github.com/7bc5587b1249f0c623e3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5053.dmg

File: libraries/entities/src/EntityItem.cpp [hunk 3]
```diff
@@ -517,12 +517,7 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
     EntityPropertyFlags propertyFlags = encodedPropertyFlags;
     dataAt += propertyFlags.getEncodedLength();
     bytesRead += propertyFlags.getEncodedLength();
-    bool useMeters = (args.bitstreamVersion >= VERSION_ENTITIES_USE_METERS_AND_RADIANS);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: your should update EntityTree.h to say you no longer support older versions...

```
    virtual bool canProcessVersion(PacketVersion thisVersion) const 
                    { return thisVersion >= VERSION_ENTITIES_SUPPORT_SPLIT_MTU; }  // we support all versions with split mtu
```

Update that to the correct minimum version

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9cb7cb314dcc0a96680586f2d21650e2da579217 available at: https://gist.github.com/7bc5587b1249f0c623e3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5053.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5053.dmg

**Pull request merged**